interpolator/interpolator.F90

```diff
@@ -3885,3841 +4125,845 @@ subroutine interp_linear ( grdin, grdout, datin, datout )
-                                                                                               !! should be linear or some other scheme for seasonal data.
-                                                                                               !! NOTIME indicates that data file has no time axis.
-
-! Flags to indicate where climatology pressure levels are pressure or sigma levels
-integer, parameter :: PRESSURE = 1, SIGMA = 2          !< Flags to indicate where climatology pressure levels are pressure or sigma levels
-
-! Flags to indicate whether the climatology units are mixing ratio (kg/kg) or column integral (kg/m2).
-! Vertical interpolation scheme requires mixing ratio at this time.
-integer, parameter :: NO_CONV = 1, KG_M2 = 2          !< Flags to indicate whether the climatology units are mixing ratio (kg/kg) or column integral (kg/m2).
-                                                                 !< Vertical interpolation scheme requires mixing ratio at this time.
-
-! Flags to indicate what to do when the model surface pressure exceeds the  climatology surface pressure level.
-integer, parameter, public :: CONSTANT = 1, ZERO = 2          !< Flags to indicate what to do when the model surface pressure
-                                                                           !< exceeds the  climatology surface pressure level.
-
-! Flags to indicate the type of vertical interpolation
-integer, parameter, public :: INTERP_WEIGHTED_P = 10, INTERP_LINEAR_P = 20, INTERP_LOG_P = 30     !< Flags to indicate the type of vertical interpolation
-!--lwh
-
-integer :: num_clim_diag = 0                                                       !< No description
-character(len=64) :: climo_diag_name(max_diag_fields)                         !< No description
-integer :: climo_diag_id(max_diag_fields), hinterp_id(max_diag_fields)     !< No description
-real ::  missing_value = -1.e10                                                       !< No description
-! sjs integer :: itaum, itaup
-
-#ifdef ENABLE_QUAD_PRECISION
-! Higher precision (kind=16) for grid geometrical factors:
- integer, parameter:: f_p = selected_real_kind(20)     !< Higher precision (kind=16) for grid geometrical factors
-#else
-! 64-bit precision (kind=8)
- integer, parameter:: f_p = selected_real_kind(15)     !< 64-bit precision (kind=8)
-#endif
-
-logical :: read_all_on_init = .false.          !< No description
-integer :: verbose = 0                              !< No description
-logical :: conservative_interp = .true.          !< No description
-logical :: retain_cm3_bug = .true.               !< No description
-
-namelist /interpolator_nml/    &
-                             read_all_on_init, verbose, conservative_interp, retain_cm3_bug
-
-contains
+do i = 1, ndim
+  call mpp_get_atts(axes(i), name=name,len=len,units=units,  &
+                    calendar=file_calendar, sense=sense)
+  select case(name)
```

Comment:
Making this `case(trim(name))` would be safer if possible.

Reply:
Good idea. Thank you.